### PR TITLE
GG-24821: Add sending cache names and sql metadata to GMC.

### PR DESCRIPTION
**Pull request by smoldachev**

### CODE REVIEW

File: modules/gmc-agent/src/main/java/org/gridgain/service/CacheService.java [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+     */
+    public CacheService(GridKernalContext ctx, WebSocketManager mgr) {
+        this.ctx = ctx;
+        this.mgr = mgr;
+        log = ctx.log(CacheService.class);
+
+        GridEventStorageManager evtMgr = ctx.event();
+
+        // Listener for cache metadata change.
+        evtMgr.addLocalEventListener(this::onDiscoveryCustomEvent, EVT_DISCOVERY_CUSTOM_EVT);
```
Comment (nva) [inline modules/gmc-agent/src/main/java/org/gridgain/service/CacheService.java]: I think it is not save to use private api. Please use org.apache.ignite.IgniteEvents#localListen

File: modules/gmc-agent/src/main/java/org/gridgain/service/CacheService.java [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+     * @param evt Event.
+     */
+    private void onCacheEvents(Event evt) {
+        sendCacheInfo();
+    }
+
+    /**
+     * @param evt Event.
+     */
+    private void onDiscoveryCustomEvent(Event evt) {
```
Comment (nva) [inline modules/gmc-agent/src/main/java/org/gridgain/service/CacheService.java]: As I remember you should return true if you want to continue receive events to listener

File: modules/gmc-agent/src/main/java/org/gridgain/service/CacheService.java (hunk 1 above)
Comment (smoldachev) [reply modules/gmc-agent/src/main/java/org/gridgain/service/CacheService.java]: Changed.

File: modules/gmc-agent/src/main/java/org/gridgain/service/CacheService.java (hunk 2 above)
Comment (smoldachev) [reply modules/gmc-agent/src/main/java/org/gridgain/service/CacheService.java]: It's true only for org.apache.ignite.IgniteEvents#localListen :)

File: modules/gmc-agent/src/main/java/org/gridgain/utils/QueryUtils.java [hunk 3]
```diff
@@ -371,4 +382,104 @@ public static String binaryToString(BinaryObject obj) {
...
+
+    /**
+     * @param cacheName Cache name.
+     * @param types Types.
+     * @return Cache sql metadata
+     */
+    public static CacheSqlMetadata queryTypesToMetadata(String cacheName, Collection<GridQueryTypeDescriptor> types) {
+        CacheSqlMetadata metadata = new CacheSqlMetadata().setCacheName(cacheName);
+
+        for (GridQueryTypeDescriptor type : types) {
```
Comment (akuznetsov-gridgain) [inline modules/gmc-agent/src/main/java/org/gridgain/utils/QueryUtils.java]: Please add comment that this code was copy-pasted from SQL internals.

File: modules/gmc-agent/src/test/java/org/gridgain/AbstractGridWithAgentTest.java [hunk 4]
```diff
@@ -31,6 +43,17 @@
 import org.springframework.test.annotation.DirtiesContext;
 import org.springframework.test.context.junit4.SpringRunner;
 
+import java.util.Collections;
```
Comment (akuznetsov-gridgain) [inline modules/gmc-agent/src/test/java/org/gridgain/AbstractGridWithAgentTest.java]: Please sort imports.

File: modules/gmc-agent/src/main/java/org/gridgain/service/CacheService.java [hunk 5]
```diff
@@ -0,0 +1,171 @@
...
+        Map<String, DynamicCacheDescriptor> cacheDescriptors = cacheProc.cacheDescriptors();
+        List<CacheInfo> cachesInfo = new ArrayList<>(cacheDescriptors.size());
+
+        for (Map.Entry<String, DynamicCacheDescriptor> item : cacheDescriptors.entrySet()) {
+            DynamicCacheDescriptor cd = item.getValue();
+
+            cachesInfo.add(
+                new CacheInfo()
+                    .setName(item.getKey())
+                    .setDeploymentId(cd.deploymentId().globalId())
```
Comment (nva) [inline modules/gmc-agent/src/main/java/org/gridgain/service/CacheService.java]:
```suggestion
                    .setDeploymentId(cd.deploymentId())
```

File: modules/gmc-agent/src/main/java/org/gridgain/utils/QueryUtils.java [hunk 6]
```diff
@@ -371,4 +382,104 @@ public static String binaryToString(BinaryObject obj) {
...
                 "typeId", obj.type().typeId(), true);
     }
+
+    /**
+     * @param cacheName Cache name.
+     * @param types Types.
+     * @return Cache sql metadata
+     */
+    public static CacheSqlMetadata queryTypesToMetadata(String cacheName, Collection<GridQueryTypeDescriptor> types) {
+        CacheSqlMetadata metadata = new CacheSqlMetadata().setCacheName(cacheName);
```
Comment (nva) [inline modules/gmc-agent/src/main/java/org/gridgain/utils/QueryUtils.java]: Missing `sqlSchema` field

File: modules/gmc-agent/src/main/java/org/gridgain/utils/QueryUtils.java (hunk 3 above)
Comment (smoldachev) [reply modules/gmc-agent/src/main/java/org/gridgain/utils/QueryUtils.java]: Done.

File: modules/gmc-agent/src/test/java/org/gridgain/AbstractGridWithAgentTest.java (hunk 4 above)
Comment (smoldachev) [reply modules/gmc-agent/src/test/java/org/gridgain/AbstractGridWithAgentTest.java]: Done.

File: modules/gmc-agent/src/main/java/org/gridgain/service/CacheService.java (hunk 5 above)
Comment (smoldachev) [reply modules/gmc-agent/src/main/java/org/gridgain/service/CacheService.java]: Done.

File: modules/gmc-agent/src/main/java/org/gridgain/utils/QueryUtils.java (hunk 6 above)
Comment (smoldachev) [reply modules/gmc-agent/src/main/java/org/gridgain/utils/QueryUtils.java]: Done.

**Pull request merged**